src/topology/algebra/field.lean

```diff
@@ -1,18 +1,118 @@
 
 /-!
-A topological field is usually described via
-`{𝕜 : Type*} [field 𝕜] [topological_space 𝕜] [topological_ring 𝕜] [has_continuous_inv 𝕜]`.
+# Topological fields
+
+A topological division ring is a topological ring whose inversion function is continuous at every
+non-zero element.
 
-The only construction in this file doesn't need to assume `[has_continuous_inv 𝕜]`.
 -/
 
+
+namespace topological_ring
+open topological_space function
+variables (R : Type*) [ring R]
+
+variables  [topological_space R]
+
+/-- The induced topology on units of a topological ring.
+This is not a global instance since other topologies could be relevant. Instead there is a class
+`induced_units` asserting that something equivalent to this construction holds. -/
+def topological_space_units : topological_space (units R) := induced (coe : units R → R) ‹_›
+
+/-- Asserts the topology on units is the induced topology. -/
+class induced_units [t : topological_space $ units R] : Prop :=
+(top_eq : t = induced (coe : units R → R) ‹_›)
+
+variables [topological_space $ units R]
+
+lemma units_topology_eq [induced_units R] :
+  ‹topological_space (units R)› = induced (coe : units R → R) ‹_› :=
+induced_units.top_eq
+
+lemma induced_units.continuous_coe [induced_units R] : continuous (coe : units R → R) :=
+(units_topology_eq R).symm ▸ continuous_induced_dom
+
+lemma units_embedding [induced_units R] :
+  embedding (coe : units R → R) :=
+{ induced := units_topology_eq R,
+  inj := λ x y h, units.ext h }
+
+instance top_monoid_units [topological_ring R] [induced_units R] :
+  has_continuous_mul (units R) :=
+⟨begin
+  let mulR := (λ (p : R × R), p.1*p.2),
+  let mulRx := (λ (p : units R × units R), p.1*p.2),
+  have key : coe ∘ mulRx = mulR ∘ (λ p, (p.1.val, p.2.val)), from rfl,
+  rw [continuous_iff_le_induced, units_topology_eq R, prod_induced_induced,
+      induced_compose, key, ← induced_compose],
+  apply induced_mono,
+  rw ← continuous_iff_le_induced,
+  exact continuous_mul,
+end⟩
+end topological_ring
+
+variables (K : Type*) [division_ring K] [topological_space K]
+
+/-- A topological division ring is a division ring with a topology where all operations are
+    continuous, including inversion. -/
+class topological_division_ring extends topological_ring K : Prop :=
+(continuous_inv : ∀ x : K, x ≠ 0 → continuous_at (λ x : K, x⁻¹ : K → K) x)
+
+namespace topological_division_ring
+open filter set
+/-!
+In this section, we show that units of a topological division ring endowed with the
+induced topology form a topological group. These are not global instances because
+one could want another topology on units. To turn on this feature, use:
+
+```lean
+local attribute [instance]
+topological_ring.topological_space_units topological_division_ring.units_top_group
+```
+-/
+
+local attribute [instance] topological_ring.topological_space_units
+
+lemma induced_units : topological_ring.induced_units K := ⟨rfl⟩
+
+local attribute [instance] induced_units
```

Comment:
This may as well be a global instance

Reply:
No, that's not always the mathematically relevant choice of topology here.

Reply:
That's not what this instance is though. This instance says "`topological_space_units` is the induced topology". It doesn't instruct lean to _use_ the induced topology, it just tells it "if you see this topology appear in a term anywhere, here's how to find the `induced_units` instance".